### DISCUSSION TITLE
vars and secrets from environment is not being populated in actions

### DISCUSSION THREAD

**Discussion by Alexnortung** [community/community#49608] [Actions] [answered]
### Select Topic Area

Bug

## Context

I have tried setting up a GitHub actions workflow in a private repository in an organization that I have admin access to. It should deploy one of our applications, and based on which branch you are pushing to it should deploy to either staging or produciton.

## Problem

I tried using the environment secrets and variables, but whenever my workflow is running the variables from the environments are not populated or they are just null whenever I try to use them with `${{ vars.MY_VAR }}`. I did try using repository variables instead and they worked fine.

I have even made a small public repo with some environment variables to test the action. It still doesn't work.

The reproducible repo is: https://github.com/Oak-Digital/actions-test
The workflow that only gets repository and not environment variables https://github.com/Oak-Digital/actions-test/actions/runs/4373248450/jobs/7651313741
and the workflow file is here: https://github.com/Oak-Digital/actions-test/blob/staging/.github/workflows/build-deploy.yml

I strongly believe this is a bug or else it is not very clear what the environment variables and secrets are for.

**Comment by KamalAman** [comment, marked as answer]
You have to specify `environment: xx` in the body of the job

https://docs.github.com/en/actions/deployment/about-deployments/deploying-with-github-actions#using-concurrency

**Reply by mattias-wiberg** [reply]
This still does not work.
name: Show Me the S3cr3tz
on: [push]
concurrency: production

jobs:
  debug:
    name: Debug
    runs-on: ubuntu-latest
    environment: production

    steps:
      - name: Check out code
        uses: actions/checkout@v2
      - name: Set env as secret
        env:
          FIREBASE_PROJECT_ID: ${{ secrets.FIREBASE_PROJECT_ID }}
          TEST: ${{ vars.TEST }}
        run: |
          import os
          for q in (os.getenv("FIREBASE_PROJECT_ID")):
            print(q)
          for q in (os.getenv("TEST")):
            print(q)
        shell: python

**Reply by ClarenceL** [reply]
For anyone that found this and is still stuck, it's important to point out that `environment` does not equal `branch`.

If you create an environment, and have variables and secrets there, you need to set the environment on the `job`. For example, let's say you made an environment called `foobar`, and you have it matching to branches `develop` and `staging`, those are different, and yes environment must be set at the job level, e.g.:

```
jobs:
  build:
    name: Build and Test
    environment: foobar
    runs-on: ubuntu-latest
```

**Reply by NicoMartinns** [reply]
@ClarenceL you just save my night, thanks!!! 

After a put the `environment` on the YAML, the secrets appeared on the workflow!

**Reply by mostafaznv** [reply]
Bro you just saved my life @ClarenceL

**Reply by YannMLD** [reply]
Thanks @ClarenceL you saved my CI
